Refresh.GameServer/Importing/TexStream.cs

```diff
@@ -0,0 +1,154 @@
+            
+            if(this._compressedChunkSizes[i] == 0) throw new FormatException("Zero sized compressed chunk found!");
+            if(this._decompressedChunkSizes[i] == 0) throw new FormatException("Zero sized decompressed chunk found!");
+        }
+
+        this._scratchBufferDecompressed = new byte[largestDecompressedChunk];
+        this._scratchBufferCompressed = new byte[largestCompressedChunk];
+        this._currentChunk = 0;
+        this._readNextChunk = true;
+
+        this._inflater = new Inflater();
+
+        this.Length = this._decompressedChunkSizes.Sum(chunk => chunk);
+    }
+
+    protected override void Dispose(bool disposing)
+    {
+        this._stream.Dispose();
+        base.Dispose(disposing);
+    }
+
+    public override void Flush() => throw new NotImplementedException();
+    public override int Read(byte[] buffer, int offset, int count)
+    {
+        //Return 0 (EOF) if we are out of chunks
+        if (this._currentChunk == this._compressedChunkSizes.Length) return 0;
+        
+        //If we need to read the next chunk
+        if (this._readNextChunk)
+        {
+            //Read the compressed data into its scratch buffer
+            this._stream.ReadAtLeast(this._scratchBufferCompressed, this._compressedChunkSizes[this._currentChunk]);
+
+            this._inflater.Reset();
+        
+            //Set the input stream to the slice of compressed scratch buffer
+            this._inflater.SetInput(this._scratchBufferCompressed, 0, this._compressedChunkSizes[this._currentChunk]);
+            //Inflate the first chunk into the decompressed scratch buffer
+            this._inflater.Inflate(this._scratchBufferDecompressed, 0, this._decompressedChunkSizes[this._currentChunk]);
+
+            this._readNextChunk = false;
+            this._chunkLeft = this._decompressedChunkSizes[this._currentChunk];
+            this._chunkRead = 0;
+        }
+        
+        Debug.Assert(this._chunkLeft > 0);
+
+        //Special case perfect or over-reads
+        if (buffer.Length >= this._chunkLeft)
+        {
+            this._scratchBufferDecompressed.AsSpan().Slice(this._chunkRead, this._chunkLeft).CopyTo(buffer);
+
+            this._readNextChunk = true;
+            this._currentChunk += 1;
+            return this._chunkLeft;
+        }
+        
+        //Copy the data it wants into the output buffer
+        this._scratchBufferDecompressed.AsSpan().Slice(this._chunkRead, buffer.Length).CopyTo(buffer);
+        
+        this._chunkRead += buffer.Length;
+        this._chunkLeft -= buffer.Length;
+        
+        return buffer.Length;
+    }
+    public override long Seek(long offset, SeekOrigin origin) => throw new NotImplementedException();
+    public override void SetLength(long value) => throw new NotImplementedException();
+    public override void Write(byte[] buffer, int offset, int count) => throw new NotImplementedException();
```

Comment:
`NotImplementedException` is typically used for methods that eventually *will* be implemented. `InvalidOperationException` is a better substitute here